src/types.rs

```diff
@@ -321,22 +321,55 @@ impl<T: ToValue + FromValue> List<T> {
     #[cfg(not(feature = "no-std"))]
     /// List as `Vec`
     pub fn to_vec(&self) -> Vec<T> {
-        let mut vec: Vec<T> = Vec::new();
-        let mut value = self.0;
-        let empty = Self::empty().0;
-        while value != empty {
-            let val = value.field(0);
-            vec.push(T::from_value(val));
-            value = value.field(1);
-        }
-        vec
+        self.iter().collect()
     }
 
     #[cfg(not(feature = "no-std"))]
     /// List as `LinkedList`
     pub fn to_linked_list(&self) -> std::collections::LinkedList<T> {
         FromValue::from_value(self.0)
     }
+
+    #[cfg(not(feature = "no-std"))]
+    /// List iterator
+    pub fn iter(&self) -> ListIterator<T> {
+        ListIterator {
+            inner: self.0,
+            _marker: PhantomData,
+        }
+    }
+}
+
+#[cfg(not(feature = "no-std"))]
```

Comment:
The `ListIterator` is a major improvement!

It looks like it will work even in a `#[no_std]` environment, so this can be removed from here and anywhere else `ListIterator` is used. You may need to specify `core::iter::Iterator`, `core::iter::IntoIter`, `core::markers::PhantomData` to make that compile.

Reply:
I saw there were a few issues with the `no-std` feature, but they are fixed on master now.